File: ffmpeg/android-ffmpeg/src/main/java/nl/bravobit/ffmpeg/Log.java
package nl.bravobit.ffmpeg;

class Log {

    private static String TAG = FFmpeg.class.getSimpleName();
    private static boolean DEBUG = false;

    public static void setDebug(boolean debug) {
        Log.DEBUG = debug;
    }

    public static void setTag(String tag) {
        Log.TAG = tag;
    }

    static void d(Object obj) {
        if (DEBUG) {
            android.util.Log.d(TAG, obj != null ? obj.toString() : "");
        }
    }

    static void e(Object obj) {
        if (DEBUG) {
            android.util.Log.e(TAG, obj != null ? obj.toString() : "");
        }
    }

    static void w(Object obj) {
        if (DEBUG) {
            android.util.Log.w(TAG, obj != null ? obj.toString() : "");
        }
    }

    static void i(Object obj) {
        if (DEBUG) {
            android.util.Log.i(TAG, obj != null ? obj.toString() : "");
        }
    }

    static void v(Object obj) {
        if (DEBUG) {
            android.util.Log.v(TAG, obj != null ? obj.toString() : "");
        }
    }

    static void e(Object obj, Throwable throwable) {
        if (DEBUG) {
            android.util.Log.e(TAG, obj != null ? obj.toString() : "", throwable);
        }
    }

    static void e(Throwable throwable) {
        if (DEBUG) {
            android.util.Log.e(TAG, "", throwable);
        }
    }

}

File: ffmpeg/android-ffmpeg/src/main/java/nl/bravobit/ffmpeg/Log.kt
package nl.bravobit.ffmpeg

import android.util.Log

internal object Log {

    private var TAG = FFmpeg::class.java.simpleName
    private var DEBUG = false

    @JvmStatic
    fun setDebug(debug: Boolean) {
        DEBUG = debug
    }

    fun setTag(tag: String) {
        TAG = tag
    }

    @JvmStatic
    fun d(obj: Any?) {
        if (DEBUG) {
            Log.d(TAG, obj?.toString() ?: "")
        }
    }

    @JvmStatic
    fun e(obj: Any?) {
        if (DEBUG) {
            Log.e(TAG, obj?.toString() ?: "")
        }
    }

    @JvmStatic
    fun w(obj: Any?) {
        if (DEBUG) {
            Log.w(TAG, obj?.toString() ?: "")
        }
    }

    @JvmStatic
    fun i(obj: Any?) {
        if (DEBUG) {
            Log.i(TAG, obj?.toString() ?: "")
        }
    }

    @JvmStatic
    fun v(obj: Any?) {
        if (DEBUG) {
            Log.v(TAG, obj?.toString() ?: "")
        }
    }

    @JvmStatic
    fun e(obj: Any?, throwable: Throwable?) {
        if (DEBUG) {
            Log.e(TAG, obj?.toString() ?: "", throwable)
        }
    }

    @JvmStatic
    fun e(throwable: Throwable?) {
        if (DEBUG) {
            Log.e(TAG, "", throwable)
        }
    }
}